react/introduction/setting_up_a_react_environment.md

```diff
@@ -0,0 +1,145 @@
+
+Once this process has completed, we should confirm that all has gone well:
+
+~~~
+$ cd my-first-react-app
+$ npm start
+~~~
+
+Provided that everything has gone according to plan, you should be greeted with a new browser tab containing a lovely spinning React logo:
+
+![Create React App Starting Page](./setting_up_a_react_environment/imgs/cra_start.png)
+
+Congratulations! You've created your first React app.
+
+### Delving Deeper
+
+Let's take a closer look at our new project. Inside you will find some folders, as well as `package.json`, `package-lock.json`, `.gitignore`, and `README.md` files. The `README.md` contains some useful information that you should take the time to skim through now.
+
+The `public` folder is where all of the static assets related to your app will go. This could include images, icons, information files for the browser, and perhaps most importantly—the `index.html` file.
+
+Inside of the `src` folder is where you will find the code that runs your app. Let's open the `index.js` file and see if we can understand what's going on:
+
+~~~js
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
+~~~
+
+Whoa! There's quite a lot here. You are not expected to recognize much of this (if any) right now. Here's a brief rundown of what is happening here:
+
+1. We import `React` itself, and its fellow `ReactDOM` package.
+2. We import some CSS styling (you may recognize this syntax from the Webpack material).
+3. We import the `App` component from `App.jsx`, so that we may place (render) it within the DOM.
+4. We import the `reportWebVitals()` function, which can be used for app analytics.
+5. We create a `root` object, and assign it an element from within our `index.html`.
+6. We invoke the `render` method which is attached to our `root` object, with some very interesting looking syntax inside the parameter.
+7. We call the function `reportWebVitals()`, at this stage this can safely be ignored.
+
+All of this may understandably look unlike anything you've seen up until now, but have no fear, once you've spent the time with this course, you'll know exactly what all of this does, and _much more_.
+
+### Keeping It Clean
+
+Another of the useful features of Create React App is that it ships with [ESLint](https://www.theodinproject.com/lessons/node-path-javascript-linting). You could also set up [Prettier](https://prettier.io/) to help keep your React code as clean as can be.
```

Comment:
We should probably link to the ESLint website rather than a lesson, thoughts?

Reply:
Agree, original thought was to keep things tied to TOP as much as possible, but with the Prettier link being an external link, it's best to keep things uniform.